app/controllers/matches_controller.rb

```diff
@@ -0,0 +1,37 @@
+class MatchesController < ApplicationController
+
+  def index
+    @id = params[:quiz]
+    if @id == nil or @id == ""
+      return redirect_to(root_url)
+    end
```

Comment:
Uma ideia de refatoração. Fica mais simples e limpo :)
```suggestion
    return redirect_to(root_url) if @id.blank?
```